frontend/src/components/elements/Progress/Progress.tsx

```diff
@@ -15,69 +15,69 @@
  * limitations under the License.
  */
 
-import React, { PureComponent, ReactNode } from "react"
+import React, { ReactElement, useEffect, useState } from "react"
 import { Map as ImmutableMap } from "immutable"
 import { Progress as UIProgress } from "reactstrap"
 
 import "./Progress.scss"
 
-export interface Props {
+export interface ProgressProps {
   width: number
   element: ImmutableMap<string, any>
 }
 
 export const FAST_UPDATE_MS = 50
 
-class Progress extends PureComponent<Props> {
-  lastValue = -1
+export default function Progress({
+  width,
+  element,
+}: ProgressProps): ReactElement {
+  const [lastValue, setLastValue] = useState(-1)
 
-  lastAnimatedTime = -1
+  const [lastAnimatedTime, setLastAnimatedTime] = useState(-1)
```

Comment:
Any reason, you chose to `useState` here? My guess is similar to stale closure setup again? The "instance variable" is not maintained between renders, so we have to use state to actually save it off? I will verify it works as expected visually.

Reply:
Yes. I thought using state would be a good approach, but open to suggestion.

Reply:
Looks like React prefers `useRef` in this case. https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

Reply:
Because of the above, we will not convert this to a hook.

Reply:
Yes, reverted Progress element